samples/quickperf/src/main/java/com/google/cloud/jdbc/quickperf/QuickPerf.java

```diff
@@ -0,0 +1,497 @@
+                                        DEFAULT_TAG));
+                            }
+
+                            long start = 0;
+                            long stop = 0;
+                            if (connection.getAutoCommit()) {
+                                // read-only
+                                start = System.nanoTime();
+                                int[] cnt = statement.executeBatch();
+                                stop = System.nanoTime() - start;
+                            } else {
+                                start = System.nanoTime();
+                                int[] cnt = statement.executeBatch();
+                                connection.commit();
+                                stop = System.nanoTime() - start;
+                            }
+
+                            batchCounter = config.getBatchSize();
+
+                            measures[batchRound] = stop / 1000000;
+                            batchRound++;
+                        }
+
+                        progress++;
+                        batchCounter--;
+                    } catch (Exception e) {
+                        if (e.getMessage().contains("ALREADY_EXISTS:")) {
+                            System.out.println("duplicate key - retry");
+                            i--;
+                        } else {
+                            throw e;
+                        }
+                    }
+                } else {
+                    System.out.println("Batching is only allowed for DML. Set batchSize=0 to disable batching.");
+                    System.exit(1);
+                }
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public float[] getMeasures() {
+        return measures;
+    }
+
+    public static Option addOption(String option, String longOption, boolean hasArgs, String desc) {
+        Option opt = new Option(option, longOption, hasArgs, desc);
+        opt.setRequired(true);
+
+        return opt;
+    }
+
+    public static Option addOption(String option, String longOption, boolean hasArgs, String desc, boolean required) {
+        Option opt = new Option(option, longOption, hasArgs, desc);
+        opt.setRequired(required);
+
+        return opt;
+    }
+
+    public static float calcPerc(float[] nums, double percentile) {
+        int n = nums.length;
+        Arrays.sort(nums);
+
+        double index = (percentile / 100) * (n - 1);
+
+        if (index == Math.floor(index)) {
+            return nums[(int) index];
+        } else {
+            int lowerIndex = (int) Math.floor(index);
+            int upperIndex = (int) Math.ceil(index);
+            float lowerValue = nums[lowerIndex];
+            float upperValue = nums[upperIndex];
+            return (float) ((1 - (index - lowerIndex)) * lowerValue + (index - lowerIndex) * upperValue);
```

Comment:
What does this do? Calculate the average between the value at lowerIndex and upperIndex?